contracts/src/instances/ezeth/EzETHBase.sol

```diff
@@ -0,0 +1,146 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.20;
+
+import { IERC20 } from "../../interfaces/IERC20.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { IRestakeManager, IRenzoOracle } from "../../interfaces/IRenzo.sol";
+import { HyperdriveBase } from "../../internal/HyperdriveBase.sol";
+import { FixedPointMath, ONE } from "../../libraries/FixedPointMath.sol";
+
+/// @author DELV
+/// @title ezETH Base Contract
+/// @notice The base contract for the ezETH Hyperdrive implementation.
+/// @dev ezETH shares are held separately in the ezETH token contract.  The
+///      value of those tokens w.r.t. ETH are found by calling the
+///      RestakeManager's calculateTVL for the total pooled ETH value and
+///      dividing by the totalSupply of ezETH.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract EzETHBase is HyperdriveBase {
+    using FixedPointMath for uint256;
+
+    /// @dev The Renzo entrypoint contract.
+    IRestakeManager internal immutable _restakeManager;
+
+    /// @dev The ezETH token contract.
+    IERC20 internal immutable _ezETH;
+
+    /// @dev The Renzo Oracle contract.
+    IRenzoOracle internal immutable _renzoOracle;
+
+    /// @dev Error for zero total supply or total pooled ether.
+    error InvalidZeroInput();
+
+    /// @notice Instantiates the ezETH Hyperdrive base contract.
+    /// @param __restakeManager The Renzo Restakemanager contract.
+    constructor(IRestakeManager __restakeManager) {
+        _restakeManager = __restakeManager;
+        _ezETH = IERC20(__restakeManager.ezETH());
+        _renzoOracle = IRenzoOracle(__restakeManager.renzoOracle());
+    }
+
+    /// Yield Source ///
+
+    /// @dev Accepts a transfer from the user in base token.
+    function _depositWithBase(
+        uint256, // _baseAmount unused
+        bytes calldata // unused
+    ) internal pure override returns (uint256, uint256) {
+        // Price calculations are too inaccurate
```

Comment:
We should update the dev comment and remove this comment. Something like this:

```
@dev This option isn't supported because the minting calculation is too imprecise.
```

I can't tell where we need to split the comment here, but make sure that we respect the 80 column width.